crates/bevy_render/src/color/mod.rs

```diff
@@ -402,107 +393,34 @@ impl Color {
-                    red,
-                    green,
-                    blue,
-                    alpha: *alpha,
-                }
-            }
+        let [red, green, blue, alpha] = self.as_rgba_f32();
+        Color::Rgba {
+            red,
+            green,
+            blue,
+            alpha,
         }
     }
 
     /// Converts a `Color` to variant `Color::RgbaLinear`
     pub fn as_rgba_linear(self: &Color) -> Color {
-        match self {
-            Color::Rgba {
-                red,
-                green,
-                blue,
-                alpha,
-            } => Color::RgbaLinear {
-                red: red.nonlinear_to_linear_srgb(),
-                green: green.nonlinear_to_linear_srgb(),
-                blue: blue.nonlinear_to_linear_srgb(),
-                alpha: *alpha,
-            },
-            Color::RgbaLinear { .. } => *self,
-            Color::Hsla {
-                hue,
-                saturation,
-                lightness,
-                alpha,
-            } => {
-                let [red, green, blue] =
-                    HslRepresentation::hsl_to_nonlinear_srgb(*hue, *saturation, *lightness);
-                Color::RgbaLinear {
-                    red: red.nonlinear_to_linear_srgb(),
-                    green: green.nonlinear_to_linear_srgb(),
-                    blue: blue.nonlinear_to_linear_srgb(),
-                    alpha: *alpha,
-                }
-            }
+        let [red, green, blue, alpha] = self.as_linear_rgba_f32();
+        Color::Rgba {
+            red,
+            green,
+            blue,
+            alpha,
         }
     }
 
     /// Converts a `Color` to variant `Color::Hsla`
     pub fn as_hsla(self: &Color) -> Color {
-        match self {
-            Color::Rgba {
-                red,
-                green,
-                blue,
-                alpha,
-            } => {
-                let (hue, saturation, lightness) =
-                    HslRepresentation::nonlinear_srgb_to_hsl([*red, *green, *blue]);
-                Color::Hsla {
-                    hue,
-                    saturation,
-                    lightness,
-                    alpha: *alpha,
-                }
-            }
-            Color::RgbaLinear {
-                red,
-                green,
-                blue,
-                alpha,
-            } => {
-                let (hue, saturation, lightness) = HslRepresentation::nonlinear_srgb_to_hsl([
-                    red.linear_to_nonlinear_srgb(),
-                    green.linear_to_nonlinear_srgb(),
-                    blue.linear_to_nonlinear_srgb(),
-                ]);
-                Color::Hsla {
-                    hue,
-                    saturation,
-                    lightness,
-                    alpha: *alpha,
-                }
-            }
-            Color::Hsla { .. } => *self,
+        let [red, green, blue, alpha] = self.as_hsla_f32();
+        Color::Rgba {
```

Comment:
This seems wrong; this was previously returning an HSLA variant, as the function name says.

Reply:
Fixed in https://github.com/bevyengine/bevy/pull/5920/commits/7720f5f7a6ad73cb8746d613e99d272c222634cf. Good catch